blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServerBuilder.scala

```diff
@@ -106,7 +110,8 @@ class BlazeServerBuilder[F[_]](
       bufferSize: Int = bufferSize,
       selectorThreadFactory: ThreadFactory = selectorThreadFactory,
       enableWebSockets: Boolean = enableWebSockets,
-      sslBits: Option[SSLConfig] = sslBits,
+      sslContext: F[Option[SSLContext]] = sslContext,
+      sslEngineConfig: SSLEngine => Unit = sslEngineConfig,
```

Comment:
This signature should make people ill, but the state is 100% internal, and it saves us some garbage on every request down here where the grickle grass grows.

Being a function lets us support the old API so people can transition gracefully.  This could be tightened up to `Option[SSLParameters]` in the long run.